rnd/autogpt_server/autogpt_server/blocks/if_block.py

```diff
@@ -0,0 +1,105 @@
+        operator: ComparisonOperator = SchemaField(
+            description="Choose the comparison operator",
+            placeholder="Select an operator",
+        )
+        value2: Any = SchemaField(
+            description="Enter the second value for comparison",
+            placeholder="For example: 20 or 'world' or False",
+        )
+        true_value: Any = SchemaField(
+            description="(Optional) Value to output if the condition is true. If not provided, value1 will be used.",
+            placeholder="Leave empty to use value1, or enter a specific value",
+            default=None,
+        )
+        false_value: Any = SchemaField(
+            description="(Optional) Value to output if the condition is false. If not provided, value1 will be used.",
+            placeholder="Leave empty to use value1, or enter a specific value",
+            default=None,
+        )
+
+    class Output(BlockSchema):
+        result: bool = SchemaField(
+            description="The result of the condition evaluation (True or False)"
+        )
+        true_output: Any = SchemaField(
+            description="The output value if the condition is true"
+        )
+        false_output: Any = SchemaField(
+            description="The output value if the condition is false"
+        )
+
+    def __init__(self):
+        super().__init__(
+            id="715696a0-e1da-45c8-b209-c2fa9c3b0be6",
+            input_schema=IfBlock.Input,
+            output_schema=IfBlock.Output,
+            test_input={
+                "value1": 10,
+                "operator": ComparisonOperator.GREATER_THAN,
+                "value2": 5,
+                "true_value": "Greater",
+                "false_value": "Not greater",
+            },
+            test_output=[
+                ("result", True),
+                ("true_output", "Greater"),
+                ("false_output", "Not greater"),
+            ],
+        )
+
+    def run(self, input_data: Input) -> BlockOutput:
+        value1 = input_data.value1
+        operator = input_data.operator
+        value2 = input_data.value2
+        true_value = (
+            input_data.true_value if input_data.true_value is not None else value1
+        )
+        false_value = (
+            input_data.false_value if input_data.false_value is not None else value1
+        )
+
+        comparison_funcs = {
+            ComparisonOperator.EQUAL: lambda a, b: a == b,
+            ComparisonOperator.NOT_EQUAL: lambda a, b: a != b,
+            ComparisonOperator.GREATER_THAN: lambda a, b: a > b,
+            ComparisonOperator.LESS_THAN: lambda a, b: a < b,
+            ComparisonOperator.GREATER_THAN_OR_EQUAL: lambda a, b: a >= b,
+            ComparisonOperator.LESS_THAN_OR_EQUAL: lambda a, b: a <= b,
+        }
+
+        try:
+            result = comparison_funcs[operator](value1, value2)
+
+            yield "result", result
+
+            if result:
+                yield "true_output", true_value
```

Comment:
the rest of the block uses `positive` `negative` should we make it consistent? we can rename the others too if it's needed.

Reply:
Interesting question! 

I think that `Positive` and `Negative` are too ambiguous in this case, as not all conditions map clearly to a positive or negative outcome.

However I like the point you're making. I personally think perhaps `Yes` and `No` are probably the most intuitive to a non technical user.

Reply:
Do we have a consensus on this one?

Reply:
yes / no sounds great!
We can change the others to yes / no too or I can follow-up on a separate PR.